packages/langium/src/index/ast-node-locator.ts

```diff
@@ -0,0 +1,60 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { LangiumDocument } from '../documents/document';
+import { findAssignment, isArray } from '../grammar/grammar-util';
+import { AstNode } from '../syntax-tree';
+
+export interface AstNodePathComputer {
+    astNodePath(node: AstNode): string;
+}
+
+export interface AstNodeLocator {
+    astNode(document: LangiumDocument, path: string): AstNode | undefined;
+}
+export class DefaultAstNodeLocator implements AstNodeLocator, AstNodePathComputer {
+    protected segmentSeparator = '/';
+
+    astNodePath(node: AstNode): string {
+        let container: AstNode | undefined = node.$container;
+        const path: string[] = [''];
+        while (container) {
+            path.push(this.pathSegment(node, container));
+            node = container;
+            container = container.$container;
+        }
+        return path.join(this.segmentSeparator);
+    }
+
+    astNode(document: LangiumDocument, path: string): AstNode | undefined {
+        const segments = path.split(this.segmentSeparator);
+        return segments.reduce((previousValue, currentValue) => {
+            if(currentValue.length === 0)
+                return previousValue;
+            const propertyIndx = currentValue.split('@');
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const value = (previousValue as any)[propertyIndx[0]];
+            return (propertyIndx.length === 2) ? value[propertyIndx[1]] : value;
+        }, document.parseResult?.value);
+    }
+
+    protected pathSegment(node: AstNode, container: AstNode): string {
```

Comment:
WDYT about supporting XPath here? The current syntax looks a lot like Xtext, I would argue for something more web-native.

Reply:
As the default implementation?  Maybe, if it fits all our needs. In the signature probably not. I would like to keep it `string` as I already saw projects that extended at least the fragment URI part

Reply:
Yes, I wouldn't change the return type, but maybe using the XPath schema would improve interaction with third-party libraries. It's just a thought though, I'm fine with the current implementation.